### PR TITLE
Openlitespeed Backport Refactor

### PR DESCRIPTION
**Pull request by josephcardillo**

### CODE REVIEW

File: apps/linode-marketplace-openlitespeed-django/README.md
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/README.md:1]: readme needs to be updated :)

File: apps/linode-marketplace-openlitespeed-django/provision.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  vars_files:
+    - group_vars/linode/vars
+  tasks:
+  # generate creds
+    - name: generating sudo user and password 
+      import_tasks: ../linode_helpers/roles/sudouser/tasks/main.yml
+      
+    - name: generating openlitespeed_password
+      set_fact:
+        openlitespeed_password: "{{ lookup('password', '/dev/null length=25 chars=ascii_letters,digits') }}"
```
Comment (n0vabyte) [inline apps/linode-marketplace-openlitespeed-django/provision.yml:16]:
```suggestion
```

File: apps/linode-marketplace-openlitespeed-django/provision.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  any_errors_fatal: true
+  user: root
+  vars_files:
+    - group_vars/linode/vars
+  tasks:
+  # generate creds
+    - name: generating sudo user and password 
+      import_tasks: ../linode_helpers/roles/sudouser/tasks/main.yml
+      
+    - name: generating openlitespeed_password
```
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/provision.yml:14]: generate & writing creds in same task
 - name: writing passwords to vars
      blockinfile:
        insertafter: EOF
        path: group_vars/linode/vars
        block: | 
          openlitespeed_password: {{ lookup('password', '/dev/null length=45 chars=ascii_letters,digits') }}

File: apps/linode-marketplace-openlitespeed-django/provision.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      set_fact:
+        openlitespeed_password: "{{ lookup('password', '/dev/null length=25 chars=ascii_letters,digits') }}"
+
+    - name: writing openlitespeed_password to vars
+      lineinfile:
+        insertafter: EOF
+        path: group_vars/linode/vars
+        line: | 
+          openlitespeed_password: {{ openlitespeed_password }}
```
Comment (n0vabyte) [inline apps/linode-marketplace-openlitespeed-django/provision.yml:23]:
```suggestion
          openlitespeed_password: {{ lookup('password', '/dev/null length=25 chars=ascii_letters,digits') }}
```

File: apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/templates/conf.j2
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/templates/conf.j2:1]: change this file to a .gitkeep

File: apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+-
+# motd and creds gen
+- name: copy MOTD template to /etc/motd
+  template:
+    src: templates/motd.j2
+    dest: /etc/motd
+
+- name: writing openlitespeed creds into file
+  copy:
+    dest: /root/.credentials
```
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml:10]: creds are now located in /home/$username/.credentials

File: apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2 [hunk 5]
```diff
@@ -0,0 +1,9 @@
+*********************************************************
+Akamai Connected Cloud OpenLiteSpeed Django Marketplace App
+App URL: 
+* The sample Django site: http://{{ ansible_default_ipv4.address }}/
+* The sample Django admin page: http://{{ ansible_default_ipv4.address }}/admin
+Credentials File: /root/.credentials
```
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2:6]: creds are now located in /home/$username/.credentials

File: apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  ansible.builtin.shell: |
+    bash <(curl -sk https://raw.githubusercontent.com/litespeedtech/ls-cloud-image/master/Setup/djangosetup.sh)
+  args:
+    executable: /bin/bash
+
+- name: Regenerate password and setup Welcome Message
+  ansible.builtin.shell: |
+    bash <(curl -sk https://raw.githubusercontent.com/litespeedtech/ls-cloud-image/master/Cloud-init/per-instance.sh)
+  args:
+    executable: /bin/bash
```
Comment (n0vabyte) [inline apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/tasks/main.yml:14]: We should use `get_url` for this :)

File: apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    mode: '0600'
+    content: |
+      All log in credentials and setup instructions can be found after logging in via SSH.
```
Comment (n0vabyte) [inline apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml:13]: We should put the creds only in the sudo user's home dir

File: apps/linode-marketplace-openlitespeed-django/roles/common/tasks/main.yml
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/roles/common/tasks/main.yml:1]: Is there a reason we took out the sudo user tasks?

File: apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2 (hunk 5 above)
Comment (n0vabyte) [inline apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2:6]:
```suggestion
Credentials File: /home/{{ username}}/.credentials
```

File: deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+  # clone repo and set up ansible environment
+  git -C /tmp clone ${GIT_REPO}
+  # for a single testing branch
+  # git -C /tmp clone -b ${BRANCH} ${GIT_REPO}
+
+  # venv
+  cd ${WORK_DIR}/${MARKETPLACE_APP}
+  pip3 install virtualenv
+  #python3 -m virtualenv env
```
Comment (hmorris3293) [inline deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh:40]: remove commment

File: deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  python3 -m virtualenv /usr/local/lsws/Example/html/
+  # source env/bin/activate
```
Comment (hmorris3293) [inline deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh:42]: remove comment

**Review by n0vabyte — CHANGES_REQUESTED** [review]

File: deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh [hunk 10]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+set -e
+trap "cleanup $? $LINENO" EXIT
+
```
Comment (hmorris3293) [inline deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh:4]: simliar to above comment, Is there a reason we took out the sudo user tasks?

File: apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/handlers/main.yml
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/handlers/main.yml:1]: change this to .gitkeep

File: apps/linode-marketplace-openlitespeed-django/roles/common/tasks/ufw_rules.yml
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/roles/common/tasks/ufw_rules.yml:1]: we can remove this file

File: apps/linode-marketplace-openlitespeed-django/roles/common/tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+---
+# ufw_install
```
Comment (hmorris3293) [inline apps/linode-marketplace-openlitespeed-django/roles/common/tasks/main.yml:2]: Does openlightspeed include a firewall UI tool? if so we can remove ufw. If they don't then we can keep ufw and add the suggested ports to the ufw_rules.yml

File: deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh (hunk 10 above)
Comment (josephcardillo) [reply deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh:4]: I was trying to make it replicate how it's currently set up. I'll add it back in, though, so we can stay consistent across the board, and also so we can save the creds to /home/$USER/.credentials

File: apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/handlers/main.yml
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/handlers/main.yml:1]: Confirmed.

File: apps/linode-marketplace-openlitespeed-django/roles/common/tasks/main.yml (hunk 11 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/common/tasks/main.yml:2]: Yes. I meant to remove this! :)

File: apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml (hunk 4 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml:10]: Check.

File: apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2 (hunk 5 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2:6]: Check.

File: apps/linode-marketplace-openlitespeed-django/roles/common/tasks/ufw_rules.yml
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/common/tasks/ufw_rules.yml:1]: Removed.

File: apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/tasks/main.yml (hunk 6 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/tasks/main.yml:14]: Will do.

File: apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2 (hunk 5 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/post/templates/motd.j2:6]: Updated already.

File: apps/linode-marketplace-openlitespeed-django/provision.yml (hunk 2 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/provision.yml]: Done.

File: apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/templates/conf.j2
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/openlitespeed/templates/conf.j2:1]: Done.

File: apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml (hunk 7 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml]: Done.

File: apps/linode-marketplace-openlitespeed-django/README.md
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/README.md:1]: Done!

**Comment by josephcardillo** [comment]
OpenLiteSpeed Django one-click is ready for re-review.

File: deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+  # clone repo and set up ansible environment
+  git -C /tmp clone ${GIT_REPO}
+  # for a single testing branch
+  # git -C /tmp clone -b ${BRANCH} ${GIT_REPO}
+
+  # venv
+  cd ${WORK_DIR}/${MARKETPLACE_APP}
+  pip3 install virtualenv
+  python3 -m virtualenv /usr/local/lsws/Example/html/
```
Comment (n0vabyte) [inline deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh:44]:
```suggestion
  python3 -m virtualenv ${VHDOCROOT}/
```

File: apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+  template:
+    src: templates/motd.j2
+    dest: /etc/motd
+
+- name: writing creds into file
+  copy:
+    dest: /home/{{ username }}/.credentials
+    mode: '0600'
+    content: |
+      Web Admin password: {{ litespeed_password }}
```
Comment (n0vabyte) [inline apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml:13]: Can you verify that this is not a base64 digest. I believe that we might need to decode the string content of the var. Maybe clean up and remove `/root/.litespeed_password`.

**Review by n0vabyte — CHANGES_REQUESTED** [review]

File: deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh (hunk 12 above)
Comment (josephcardillo) [reply deployment_scripts/linode-marketplace-openlitespeed-django/openlitespeed-django-deploy.sh:44]: Resolved.

File: apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml (hunk 13 above)
Comment (josephcardillo) [reply apps/linode-marketplace-openlitespeed-django/roles/post/tasks/main.yml:13]: There's nothing here that indicates it's base64:
https://docs.litespeedtech.com/cloud/images/django/#how-to-access-the-installed-software

When logging in through http://<ip_address>:7080 to the openlitespeed dashboard, it works with the creds `admin` and the password at /root/litespeed_password

**Comment by josephcardillo** [comment]
The OLS Django app is ready for re-review. Updates made:
- Django superuser creation automated
- Config UI screens bypassed
- domainsetup.sh script automated
- SSL added for Django and OLS admin 
- /etc/pam.d/sshd motd commected out in favor or our motd

**Review by n0vabyte — APPROVED** [review]
Amazing work! This looks solid. Deployment passed and all requirements satisfied!

**Pull request merged**